tests/xgboost/test_xgboost_autolog.py

```diff
@@ -238,6 +274,53 @@ def test_xgb_autolog_logs_metrics_with_early_stopping(bst_params, dtrain):
             assert metric_history == evals_result[eval_name][metric_name] + [best_metrics]
 
 
+@pytest.mark.large
+def test_xgb_autolog_batch_metrics_logger_logs_metrics_with_early_stopping(bst_params, dtrain):
```

Comment:
During some of our offline discussion, I think we suggested `spy`ing on the `BatchMetricsLogger` so that metrics are logged to the run as well. That way, we can compare `patched_metrics_data` to `run.data.metrics` and ensure that the same keys are present. This, coupled with 1-2 concrete value checks (e.g., `"best_iteration" in metric_dict`), should be sufficient.

Can we update our test cases to follow this paradigm? For setting up spies, the `wraps` keyword argument on `patch()` should be helpful: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock (`wraps` is also defined on `patch()` but only seems to be documented here)

Reply:
EDIT: My suggestion to use `patch()` is actually misguided. Instead, we should update the mocked method to call the original method. We can do this by grabbing a reference to the original method before we apply the patch.